src/kernels/telemetry/helper.ts

```diff
@@ -68,7 +76,12 @@ export function trackKernelResourceInformation(resource: Resource, information:
     currentData.kernelLiveCount = information.kernelLiveCount || currentData.kernelLiveCount || 0;
     currentData.kernelInterpreterCount = information.kernelInterpreterCount || currentData.kernelInterpreterCount || 0;
     currentData.pythonEnvironmentCount = InterpreterCountTracker.totalNumberOfInterpreters;
-
+    if (information.userExecutedCell) {
+        currentData.userExecutedCell = true;
+    }
+    if (typeof information.disableUI === 'boolean') {
+        currentData.disableUI = information.disableUI;

```

Comment:
This way we now know whether the telemtry for a particular kernel is a result of user execution or not.
Previously it was for all kernels as we were only interested in whether things worked - at a user level.